From 4228132424d7b44e732a54756498fd5114a0909e Mon Sep 17 00:00:00 2001
From: Pavel Astakhov <pastakhov@yandex.ru>
Date: Thu, 14 Oct 2021 01:03:07 +0600
Subject: [PATCH 1/2] Update composer.json requere "ircmaxell/random-lib":
 "v1.2.0"

It fixes the error message for composer update:
`composer.json requires ircmaxell/random-lib dev-master, found ircmaxell/random-lib[dev-cipher_mixer, dev-improved_string_presets, v1.0.0, 1.0.x-dev, v1.1.0, v1.2.0] but it does not match the constraint.`
---
 composer.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/composer.json b/composer.json
index c872106..1f62159 100644
--- a/composer.json
+++ b/composer.json
@@ -13,6 +13,6 @@
 	],
     "require": {
 		"composer/installers": ">=1.0.1",
-		"ircmaxell/random-lib": "dev-master"
+		"ircmaxell/random-lib": "v1.2.0"
     }
 }

From 5d673b7901b24bb2f8dd1e5c3b9459f5575306af Mon Sep 17 00:00:00 2001
From: Pavel Astakhov <pastakhov@yandex.ru>
Date: Thu, 14 Oct 2021 21:50:47 +0600
Subject: [PATCH 2/2] Replace removed and deprecated functions (requires
 MediaWiki >= 1.35.0)

---
 SpecialMassPasswordReset.php | 14 +++++++++-----
 extension.json               |  2 +-
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/SpecialMassPasswordReset.php b/SpecialMassPasswordReset.php
index 8610434..fe0e20a 100644
--- a/SpecialMassPasswordReset.php
+++ b/SpecialMassPasswordReset.php
@@ -1,8 +1,10 @@
 <?php
 
+use MediaWiki\MediaWikiServices;
+
 /**
- * 
- * 
+ *
+ *
  */
 class SpecialMassPasswordReset extends SpecialPage {
 	public function __construct() {
@@ -10,8 +12,9 @@ public function __construct() {
 	}
 
 	/**
+	 * @inerhitDoc
 	 */
-	public function execute( $par ) {
+	public function execute( $subPage ) {
 		$this->setHeaders();
 		$request = $this->getRequest();
 		$out = $this->getOutput();
@@ -21,7 +24,8 @@ public function execute( $par ) {
 			return;
 		}
 
-		if( !in_array( 'sysop', $this->getUser()->getEffectiveGroups()) ) {
+		$ugm = MediaWikiServices::getInstance()->getUserGroupManager();
+		if( !in_array( 'sysop', $ugm->getUserEffectiveGroups( $this->getUser() ) ) ) { // TODO: Replace me with $ugm->userHasRight()
 			$out->addHTML( '<div class="errorbox">This page is only accessible by users with sysop right.</div>' );
 			return;
 		}
@@ -30,7 +34,7 @@ public function execute( $par ) {
 			'id' => 'password_reset',
 			'method' => 'post',
 			"enctype" => "multipart/form-data",
-			'action' => $this->getTitle()->getFullUrl()
+			'action' => $this->getPageTitle()->getFullUrl()
 		];
 		$out->addHTML(
 			Html::openElement( 'form', $formOpts ) . "<br>" .
diff --git a/extension.json b/extension.json
index 727c1cf..562f393 100644
--- a/extension.json
+++ b/extension.json
@@ -7,7 +7,7 @@
 	"license-name": "GPL-2.0-or-later",
 	"type": "specialpage",
 	"requires": {
-		"MediaWiki": ">= 1.31.0"
+		"MediaWiki": ">= 1.35.0"
 	},
 	"MessagesDirs": {
 		"MassPasswordReset": [
